### PR TITLE
added DQN tutorials

### PR DESCRIPTION
**Pull request by veds12**
- [x] DQN
- [x] Double DQN
- [x] Dueling DQN
- [x] NoisyNet DQN
- [x] Categorical DQN
- [x] Prioritized DQN
- [x] DDPG
- [x] TD3
- [x] SAC

### CODE REVIEW

File: docs/source/usage/tutorials/Deep/Double_DQN.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
+=====================
+Double Deep Q-Network
+=====================
+
+Objective
+=========
+
+Double DQN builds upon the notion of Double Q-Learning and extends it to Deep Q-networks. The main objective is to resolve the problem of overestimation of Q-values. This is done by defining the traget q-values in a slightly differen manner. The loss function is defined as:
```
Comment (Sharad24) [inline docs/source/usage/tutorials/Deep/Double_DQN.rst:8]: Can you explain here why is there an overestimation of Q-values in a vanilla DQN?

File: docs/source/usage/tutorials/Deep/Dueling_DQN.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
+======================
+Dueling Deep Q-Network
+======================
+
+Objective
+=========
+
+The main objective of DQN is to learn a function approximator for the Q-function using a
```
Comment (Sharad24) [inline docs/source/usage/tutorials/Deep/Dueling_DQN.rst:8]: Again we should start of why we need a Dueling DQN i.e., the motivation behind it.

File: docs/source/usage/tutorials/Deep/NoisyNet_DQN.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
+=======================
+NoisyNet Deep Q-Network
```
Comment (Sharad24) [inline docs/source/usage/tutorials/Deep/NoisyNet_DQN.rst:2]:
```suggestion
Deep Q Networks with Noisy Nets
```

File: docs/source/usage/tutorials/Deep/NoisyNet_DQN.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+=======================
+
+Objective
+=========
+
+NoisyNet DQN is a variant of DQN which uses fully connected layers with noisy parameters to drive exploration. Thus, the parametrised action-value function can now be seen as a random variable. The new loss function which
```
Comment (Sharad24) [inline docs/source/usage/tutorials/Deep/NoisyNet_DQN.rst:8]: Missing Motivation.

File: docs/source/usage/tutorials/Deep/Double_DQN.rst (hunk 1 above)
Comment (veds12) [reply docs/source/usage/tutorials/Deep/Double_DQN.rst:8]: Cool, will add this part to every tutorial

**Pull request merged**